src/Entities/Player.cpp

```diff
@@ -1447,13 +1447,16 @@ void cPlayer::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 	//  ask plugins to allow teleport to the new position.
 	if (!cRoot::Get()->GetPluginManager()->CallHookEntityTeleport(*this, m_LastPosition, Vector3d(a_PosX, a_PosY, a_PosZ)))
 	{
-		SetPosition(a_PosX, a_PosY, a_PosZ);
-		FreezeInternal(GetPosition(), false);
-		m_LastGroundHeight = a_PosY;
-		m_bIsTeleporting = true;
-
-		m_World->BroadcastTeleportEntity(*this, GetClientHandle());
-		m_ClientHandle->SendPlayerMoveLook();
+		m_World->QueueTask([this, a_PosX, a_PosY, a_PosZ](cWorld & a_World)
```

Comment:
Good point. This also means `DoMoveToWorld` may be currently unsafe.
I also found out `ScheduleMoveToWorld` is buggy when called by plugins.
I am working on a proper fix all 3 methods.

Can we merge this first? warps are currently broken and they are critical for many servers. The fix is worth the minor risk of a crash. I will be fixing this fully very soon.